### PR TITLE
chmod: avoid utf8 operation for mode parsing

### PR DESCRIPTION
**Pull request by youknowone**

### CODE REVIEW

**Comment by jhscheer** [comment]
What is the motivation here?
If things could break without this I would like to write a test for this.

File: src/uu/chmod/src/chmod.rs [hunk 1]
```diff
@@ -177,11 +177,11 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
     for arg in args {
         if arg.starts_with('-') {
-            if let Some(second) = arg.chars().nth(1) {
+            if let Some(second) = arg.as_bytes().get(1) {
                 match second {
-                    'r' | 'w' | 'x' | 'X' | 's' | 't' | 'u' | 'g' | 'o' | '0'..='7' => {
+                    b'r' | b'w' | b'x' | b'X' | b's' | b't' | b'u' | b'g' | b'o' | b'0'..=b'7' => {
                         // TODO: use strip_prefix() once minimum rust version reaches 1.45.0
-                        *arg = arg[1..arg.len()].to_string();
+                        *arg = arg[1..].to_string();
```
Comment (miDeb) [inline src/uu/chmod/src/chmod.rs:184]: I think the bigger performance improvement would be to get rid of the `.to_string()`, as that will allocate a new string.
```suggestion
                        arg.remove(0);
```

**Review by miDeb — COMMENTED** [review]
While it looks like this operation is actually safe, I don't believe we should make this change; I'd consider it a premature optimization (unless you can show an improvement in an actual benchmark, but I don't believe this will cause a significant performance improvement)

**Comment by tertsdiepraam** [comment]
With the arguments brought up by @miDeb, I think we should close this and open a new PR to remove the allocations. @youknowone Thanks for suggestion!

**Pull request closed without merging**